setInterval/3-setIntervalPolyfill.js

```diff
@@ -0,0 +1,72 @@
+const {
+    setIntervalPolyfill,
+    clearIntervalPolyfill
+}= (()=>{
+
+    // Closure Scope.
+    var intervalId = 0;
+    var intervalMap = {}
+
+    function setIntervalPolyfill (callbackFn, delay=0) {
+
+        // Throw error if callback is not a function.
+        if(typeof callbackFn !== "function"){
+            throw new TypeError('"callback" argument must be a funtion.')
+        }
+
+        var id = intervalId++;
+        intervalMap[id] = true;
+
+        // to be passed as paremeters in callback function call.
+        var args = Array.prototype.slice.call(arguments,2);
+
+        (function repeat(){
+            setTimeout(
+                ()=>{
+                    if(intervalMap[id]){
+                        callbackFn(args);
+                        repeat();
+                    }  
+                },
+                delay
+            );
+        })();
+
+        return id;
+    }
+
+
+    function clearIntervalPolyfill(intervalId) {
+        delete intervalMap[intervalId];
```

Comment:
We are clearing `id` from the `map` but `setTimeout` itself returns an `id`. Aren't we going to clear that?

Reply:
SetTimeout will return new Id everytime.
I guess it will be tough to manage that ID

Reply:
But then shouldn't we mention that? May in the video if not in the code as users might be thinking what happened to `id` returned by `setTimeout` itself. What do you think?